neo/core/analogsignal.py

```diff
@@ -612,8 +604,9 @@ def downsample(self, downsampling_factor, **kwargs):
         -----
         For resampling the signal with a fixed number of samples, see `resample` method.
         """
-
-        if not HAVE_SCIPY:
+        try:
+            import scipy.signal
```

Comment:
Import only here so cost if sonly when people want to use the function. Which I assume is rarer than importing the package overall.